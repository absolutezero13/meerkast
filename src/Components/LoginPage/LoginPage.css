.login-btn {
  width: 200px;
  height: 100px;
  border: 10px solid rgb(165, 78, 20);
  border-radius: 50px;
  color: white;
  background: black;
  font-size: 15px;
  outline: none;
  cursor: pointer;
  transition: all 0.3s ease;
}

.login-btn:active {
  transform: translateY(5%);
}
