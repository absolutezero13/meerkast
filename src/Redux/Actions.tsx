export const setTextFilter = (text: string = "") => ({
  type: "SET_TEXT_FILTER",
  text,
});
export const setSortByAscendingYear = () => ({
  type: "SORT_BY_ASCENDING_YEAR",
});
export const setSortByDescendingYear = () => ({
  type: "SORT_BY_DESCENDING_YEAR",
});
export const setSortByAscendingTitle = () => ({
  type: "SORT_BY_ASCENDING_TITLE",
});
export const setSortByDescendingTitle = () => ({
  type: "SORT_BY_DESCENDING_TITLE",
});
