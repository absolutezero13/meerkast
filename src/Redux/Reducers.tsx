import data from "../feed/sample.json"; // same with require (I guess)

const getData = (typeOfProgram: string) =>
  data.entries
    .filter(
      (entry) =>
        entry.programType === typeOfProgram && entry.releaseYear >= 2010
    )
    .slice(0, 21)
    .sort((a, b) => {
      if (a.title === b.title) {
        return 0;
      }
      if (typeof a.title === typeof b.title) {
        return a.title < b.title ? -1 : 1;
      }
      return typeof a.title < typeof b.title ? -1 : 1;
    });
export const movieData: Array<Object> = getData("movie").map((movie: any, i) => ({
  ...movie,
  id: i,
}));
export const serieData: Array<Object> = getData("series").map((serie: any, i) => ({
  ...serie,
  id: i * 50,
}));
const moviesReducerDefaultState = [...movieData];
export const moviesReducer = (state: Object = moviesReducerDefaultState) => {
  return state;
};
const seriesReducerDefaultState = [...serieData];
export const seriesReducer = (state: Object = seriesReducerDefaultState) => {
  return state;
};
type Action = {
  type: string;
  text?: string;
};
const filterReducersDefaultState = {
  text: "",
  sortBy: "",
};
export const filterReducer = (
  state: Object = filterReducersDefaultState,
  action: Action
) => {
  switch (action.type) {
    case "SET_TEXT_FILTER":
      return { ...state, text: action.text };
    case "SORT_BY_ASCENDING_YEAR":
      return { ...state, sortBy: "ascending-year" };
    case "SORT_BY_DESCENDING_YEAR":
      return { ...state, sortBy: "descending-year" };
    case "SORT_BY_ASCENDING_TITLE":
      return { ...state, sortBy: "ascending-title" };
    case "SORT_BY_DESCENDING_TITLE":
      return { ...state, sortBy: "descending-title" };
    default:
      return state;
  }
};

const obj: { [k: string]: number } = {};
