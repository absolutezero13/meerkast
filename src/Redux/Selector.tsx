export const getVisibleShows = (showData: any, { text, sortBy }: any) => {
  return showData
    .filter((show: any) => {
      if (text.length >= 3) {
        const textMatch = show.title.toLowerCase().includes(text.toLowerCase());
        return textMatch;
      } else {
        return showData;
      }
    })
    .sort((a: any, b: any) => {
      if (sortBy === "ascending-year") {
        return a.releaseYear > b.releaseYear ? 1 : -1;
      } else if (sortBy === "descending-year") {
        return a.releaseYear < b.releaseYear ? 1 : -1;
      } else if (sortBy === "ascending-title") {
        return a.title > b.title ? 1 : -1;
      } else if (sortBy === "descending-title") {
        return a.title < b.title ? 1 : -1;
      } else {
        return a + b;
      }
    });
};
