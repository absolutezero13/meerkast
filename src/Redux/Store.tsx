import { createStore, combineReducers, applyMiddleware, compose } from "redux";
import { moviesReducer, filterReducer, seriesReducer } from "./Reducers";
import thunk from "redux-thunk";
import authReducer from "./AuthReducer";
declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
  }
}
const composeEnhancers =
  window["__REDUX_DEVTOOLS_EXTENSION_COMPOSE__"] || compose;

export const store = createStore(
  combineReducers({
    movies: moviesReducer,
    series: seriesReducer,
    filters: filterReducer,
    auth: authReducer,
  }),
  composeEnhancers(applyMiddleware(thunk))
);
