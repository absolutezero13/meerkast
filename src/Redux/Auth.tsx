import {
  firebase,
  googleAuthProvider,
  facebookAuthProvider,
} from "../firebase/firebase";
import { store } from "./Store";
export const login = (uid: string) => ({
  type: "LOGIN",
  uid,
});
export const logout = () => ({
  type: "LOGOUT",
});
export const startLogin = () => {
  return () => {
    return firebase.auth().signInWithPopup(googleAuthProvider);
  };
};
export const startLoginFacebook = () => {
  return () => {
    return firebase.auth().signInWithPopup(facebookAuthProvider);
  };
};

export const startLogout = () => {
  return () => {
    return firebase.auth().signOut();
  };
};
