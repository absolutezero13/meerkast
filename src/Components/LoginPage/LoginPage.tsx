import React from "react";
import "./LoginPage.css";
import { connect } from "react-redux";
import { startLogin } from "../../Redux/Auth";

interface loginProps {
  startLogin: any;
}
const LoginPage: React.FC<loginProps> = ({ startLogin }) => {
  return (
    <div className="body">
      <button onClick={startLogin} className="login-btn">
        Log in with Google
      </button>
    </div>
  );
};
const mapDispatchToProps = (dispatch: any) => ({
  startLogin: () => dispatch(startLogin()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
