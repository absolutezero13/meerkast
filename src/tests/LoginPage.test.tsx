import React from "react";
import { render } from "@testing-library/react";
import LoginPage from "../Components/LoginPage/LoginPage";
import { Provider } from "react-redux";
import { store } from "../Redux/Store";
import { shallow } from "enzyme";
test("should match the snaphot", () => {
  const { container } = render(
    <Provider store={store}>
      <LoginPage />
    </Provider>
  );
  expect(container).toMatchSnapshot();
});

// test("button should call startlogin func", () => {
//   const startLogin = jest.fn();
//   const container = shallow(
//     <Provider store={store}>
//       <LoginPage startLogin={startLogin} />
//     </Provider>
//   );
//   container.find("button").simulate("click").expect(startLogin).toHave;
// });
