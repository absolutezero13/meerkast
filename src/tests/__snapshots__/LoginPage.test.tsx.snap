// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snaphot 1`] = `
<div>
  <div
    class="body"
  >
    <button
      class="login-btn"
    >
      Log in with Google
    </button>
  </div>
   
</div>
`;
